### PR TITLE
Fix error when adding a price while a price already exists for previous years

### PR DESCRIPTION
**Pull request by Thource**
Add effectiveYear to price auditing

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/JourneyQueryRepository.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ class JourneyQueryRepository(@Autowired val jdbcTemplate: JdbcTemplate) {
 
     val placeholders =
       (listOf(supplier.name, fromSiteName, toSiteName).filter { !it.isNullOrBlank() } + effectiveYear).toTypedArray()
-    return jdbcTemplate.query(selectPricesSQL, placeholders, pricesRowMapper)
+    return jdbcTemplate.query(selectPricesSQL, pricesRowMapper, *placeholders)
```
Comment (Thource) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/move/JourneyQueryRepository.kt:58]: The previous arg order has been deprecated and replaced with this one

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/spreadsheet/PricesSpreadsheetGenerator.kt [hunk 2]
```diff
@@ -65,7 +65,7 @@ class PricesSpreadsheetGenerator(
         logger.info("Adding $supplier prices for effective year ${effectiveYearForDate(startDate)}.")
       }.apply { writePrices(supplierPrices.get(supplier, effectiveYearForDate(startDate))) }
 
-      return createTempFile(suffix = "xlsx").apply {
+      return File.createTempFile("tmp", "xlsx").apply {
```
Comment (Thource) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/spreadsheet/PricesSpreadsheetGenerator.kt:68]: `createTempFile` was deprecated, deprecation warning advised using this instead

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/SupplierPricingService.kt [hunk 3]
```diff
@@ -19,20 +19,32 @@ class SupplierPricingService(
...
-      throw RuntimeException("Supplier $supplier price already exists from ${fromLocation.siteName} to ${toLocation.siteName}")
-    }
+    priceRepository.findBySupplierAndFromLocationAndToLocationAndEffectiveYear(
+      supplier,
+      fromLocation,
+      toLocation,
+      effectiveYear
+    )
+      ?.let {
+        throw RuntimeException("Supplier $supplier price already exists from ${fromLocation.siteName} to ${toLocation.siteName}")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/SupplierPricingService.kt:37]: Perhaps include the effective year in the exception message.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**